CookSmart/CookSmart/UI/IngredientListView.swift

```diff
@@ -0,0 +1,263 @@
+            },
+            label: {
+              Image(systemName: "xmark")
+                .foregroundStyle(CSColor.accent.asSwiftUIColor())
+            }
+          )
+        }
+        ToolbarItem(placement: .principal) {
+          Text("Ingredients").csTextStyle(.heading)
+        }
+        ToolbarItem(placement: .topBarTrailing) {
+          NavigationLink(
+            destination: EditIngredientViewWithToolbar(),
+            label: {
+              Image(systemName: "plus")
+                .foregroundStyle(CSColor.accent.asSwiftUIColor())
+            }
+          )
+        }
+        ToolbarItem(placement: .bottomBar) {
+          Button(action: {
+            isShowingResetAlert = true
+          }, label: {
+            Text("Reset to Defaults").csTextStyle(.plainButton)
+          })
+        }
+      }
+      .alert("Are you sure?", isPresented: $isShowingResetAlert) {
+        Button("Reset", role: .destructive) {
+          isShowingResetAlert = false
+          // O&A Question: Why does this not trigger a nice animation?
+          ingredientsStore.resetToDefault()
+        }
+      } message: {
+        Text("Resetting to defaults will remove all your added and edited ingredients")
+      }
+    }
+  }
+}
+
+struct IngredientGroupSection: View {
+  let group: any IngredientGroup
+  weak var delegate: IngredientListViewDelegate?
+
+  var body: some View {
+    Section(header: Text(group.name).csTextStyle(.subheading)) {
+      ForEach(group.ingredients) { ingredient in
+        IngredientListViewCell(
+          ingredient: ingredient,
+          delegate: delegate
+        )
+      }.onDelete { indexSet in
+        // O&A Question: the animation is not great when an ingredient is removed that's both in recents
+        // and in one of the groups. Is there a smoother way to do this?
+        let groupIngrs = group.ingredients
+        IngredientsStore.shared.delete(ingredientsWithIds: indexSet.map { groupIngrs[$0].id })
+      }
+    }
+  }
+}
+
+struct IngredientListViewCell: View {
+  var ingredient: Ingredient!
+  weak var delegate: IngredientListViewDelegate?
+
+  var body: some View {
+    HStack {
+      HStack {
+        Text(ingredient.name).csTextStyle(.coreContent)
+        Spacer()
+      }
+      .contentShape(Rectangle())
+      .onTapGesture {
+        self.delegate?.ingredientListViewSelected(
+          ingredientAtFlattenedIndex: UInt(IngredientsStore.shared.flattenedForIngredient(withId: ingredient.id))
+        )
+      }
+      Image(systemName: "info.circle")
+        .foregroundStyle(CSColor.accent.asSwiftUIColor())
+        // O&A Question: Do you guys know of a less hacky way to accomplish something seemingly so basic.
+        // Even with this, the tap hit box for editing ingredients is still not perfect :(
+        // Also, it turns out that for some reason long-tapping anywhere in the cell triggers this action :(
+        .overlay(
+          NavigationLink(
```

Comment:
idk, again, I've never used NavigationLink. All the navigation I've done has been custom Airbnb stuff, so idk what we have out of the box.